### DISCUSSION TITLE
Help with WZDx v4.1 GeoJSON validation

### DISCUSSION THREAD

**Discussion by persjabari** [usdot-jpo-ode/jpo-wzdx#343] [Q&A] [answered]
I have been having problems validating a WZDx v4.1 feed that I have been involved with developing. I have attempted to use both the Python `jsonschema` library and https://www.jsonschemavalidator.net/, and validation fails with what appears to be a false negative.

Below is the error that I get with my custom validator script that uses the `jsonschema` library:
```
jsonschema.exceptions.ValidationError: 'detour' was expected

Failed validating 'const' in schema[1]['allOf'][0]['properties']['core_details']['properties']['event_type']:
    {'const': 'detour'}

On instance['core_details']['event_type']:
    'work-zone'
```

https://www.jsonschemavalidator.net/ fails validation with the same error. Validation also fails, with the same error, for example feeds posted in https://github.com/usdot-jpo-ode/wzdx/tree/main/examples/WorkZoneFeed. Note that the feed I have developed only has `WorkZoneRoadEvent` entities, and all event types are being correctly set to `work-zone`, as the specification instructs.

I would appreciate any guidance on how to correctly validate a WZDx v4.1 GeoJSON feed. Thanks.

**Comment by j-d-b** [comment, marked as answer]
@persjabari thanks for raising this issue. I tried running every Work Zone Feed example against the JSON schema and they all failed, though it was due to issues with the examples. All example files in some cases incorrectly provide strings for properties that must be boolean values. I fixed the example file issues in #344, awaiting approval.

The error message about "const" is misleading—at least in the examples case there were other issues that were also shown and resolving those removed the "const" message. I was using https://www.jsonschemavalidator.net/; does it provide other error messages for the feed you are trying to validate in addition to the "const" message?

**Reply by persjabari** [reply]
Thanks for the response @j-d-b. You are correct. Looking more closely at the errors on https://www.jsonschemavalidator.net/, the `detour` const message is indeed misleading. My feed had some problems with timestamps, and fixing those resolved the issue with the `detour` const.

Unfortunately, the `jsonschema` Python library reports only the first "error" it encounters which happens to be the `detour` const issue; it is not as easy to debug a feed using `jsonschema`.
